### PR TITLE
Scheduled Updates: Move health check functions into hooks

### PR DESCRIPTION
**Pull request by ouikhuan**
Fixes https://github.com/Automattic/dotcom-forge/issues/6700

## Proposed changes:

This is a follow-up from #36990
* Add Scheduled_Updates_Health_Paths::update to jetpack_scheduled_update_created action.
* Add Scheduled_Updates_Health_Paths::clear to jetpack_scheduled_update_deleted action.
* Remove original functions from `WPCOM_REST_API_V2_Endpoint_Update_Schedules`

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Follow #36990 and create schedules using REST API Console with health check paths and everything should work as before.

### CODE REVIEW

File: projects/packages/scheduled-updates/src/class-scheduled-updates.php [hunk 1]
```diff
@@ -61,6 +61,8 @@ public static function init() {
 		add_action( 'jetpack_scheduled_update_deleted', array( __CLASS__, 'enable_autoupdates' ), 10, 2 );
 		add_action( 'jetpack_scheduled_update_updated', array( Scheduled_Updates_Logs::class, 'replace_logs_schedule_id' ), 10, 2 );
 		add_action( 'jetpack_scheduled_update_deleted', array( Scheduled_Updates_Logs::class, 'delete_logs_schedule_id' ), 10, 3 );
+		add_action( 'jetpack_scheduled_update_created', array( Scheduled_Updates_Health_Paths::class, 'update' ), 10, 3 );
+		add_action( 'jetpack_scheduled_update_deleted', array( Scheduled_Updates_Health_Paths::class, 'clear' ), 10, 1 );
```
Comment (obenland) [inline projects/packages/scheduled-updates/src/class-scheduled-updates.php]: 10 and 1 are default values for priority and accepted arguments and could be omitted.

File: projects/packages/scheduled-updates/src/class-scheduled-updates-health-paths.php [hunk 2]
```diff
@@ -36,11 +36,14 @@ public static function get( $schedule_id ) {
...
 	 *
-	 * @param string $schedule_id Request ID.
-	 * @param array  $paths       List of paths to save.
+	 * @param string           $schedule_id The ID of the schedule.
+	 * @param object           $event       The event object.
+	 * @param \WP_REST_Request $request     The request object.
 	 * @return bool
 	 */
-	public static function update( $schedule_id, $paths ) {
+	public static function update( $schedule_id, $event, $request ) {
```
Comment (obenland) [inline projects/packages/scheduled-updates/src/class-scheduled-updates-health-paths.php]: I don't know if we want to keep `update()` as a library function or not, I don't have a preference either way.
If we do, it's fine to create a separate hook callback that just calls this function with the right parameters.

**Review by obenland — DISMISSED** [review]

File: projects/packages/scheduled-updates/src/class-scheduled-updates.php (hunk 1 above)
Comment (ouikhuan) [reply projects/packages/scheduled-updates/src/class-scheduled-updates.php]: Thanks, resolved!

File: projects/packages/scheduled-updates/src/class-scheduled-updates-health-paths.php (hunk 2 above)
Comment (ouikhuan) [reply projects/packages/scheduled-updates/src/class-scheduled-updates-health-paths.php]: Thanks for the suggestion, I decided to keep it and create a separate hook to handle this.

File: projects/packages/scheduled-updates/src/class-scheduled-updates.php [hunk 3]
```diff
@@ -483,4 +485,18 @@ private static function file_system_write_access() {
...
+	 * @param string           $id      The ID of the schedule.
+	 * @param object           $event   The event object.
+	 * @param \WP_REST_Request $request The request object.
+	 * @return bool
+	 */
+	public static function updates_health_paths( $id, $event, $request ) {
+		$schedule = $request['schedule'];
+		$paths    = $schedule['health_check_paths'] ?? array();
+		return Scheduled_Updates_Health_Paths::update( $id, $paths );
+	}
```
Comment (obenland) [inline projects/packages/scheduled-updates/src/class-scheduled-updates.php]: Let's move this into the `Scheduled_Updates_Health_Paths` class to keep that functionality together

**Review by obenland — DISMISSED** [review]

File: projects/packages/scheduled-updates/src/class-scheduled-updates.php (hunk 3 above)
Comment (ouikhuan) [reply projects/packages/scheduled-updates/src/class-scheduled-updates.php]: Sure, moved!

**Review by zaerl — APPROVED** [review]
Work like a charm, great work.

Screenshot 2024-04-29 alle 09 13 32

image

**Pull request merged**